homeassistant/components/epson/__init__.py

```diff
@@ -3,32 +3,35 @@
 import logging
 
 from epson_projector import Projector
-from epson_projector.const import POWER, STATE_UNAVAILABLE as EPSON_STATE_UNAVAILABLE
+from epson_projector.const import (
+    PWR_OFF_STATE,
+    STATE_UNAVAILABLE as EPSON_STATE_UNAVAILABLE,
+)
 
 from homeassistant.components.media_player import DOMAIN as MEDIA_PLAYER_PLATFORM
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.const import CONF_HOST
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from .const import DOMAIN
-from .exceptions import CannotConnect
+from .exceptions import CannotConnect, PoweredOff
 
 PLATFORMS = [MEDIA_PLAYER_PLATFORM]
 
 _LOGGER = logging.getLogger(__name__)
```

Comment:
This PR changes config-flow data scheme, but it provides no migration to newer version.

We should increment config flow data version to 2 and add the migration like this:
```python
async def async_migrate_entry(_, config_entry: ConfigEntry):
    """Migrate old entry."""
    _LOGGER.debug("Migrating from version %s", config_entry.version)

    if config_entry.version == 1:
        new = {**config_entry.data, CONF_PROTOCOL: PROTO_HTTP} # customize for this case
        config_entry.data = {**new}
        config_entry.version = 2

    _LOGGER.info("Migration to version %s successful", config_entry.version)
    return True
```

Reply:
old config flow didn't have proper unique id set and I don't think it can be migrated as it won't recognize old config...

Reply:
It's about the port entry. All instances with current config flow have the port option set. During migration, you should remove that option from the list and bump version number.

Reply:
ok, I will do that

Reply:
I tried to implement migration and I don't see anyway to do it properly.
I need serial number of projector in new config and I can get it only while projector is turned on.
In migration process I don't have a way to ask user to turn on projector and force migration will just extend errors during HA restart process.

Reply:
Does it mean this is a breaking change, and new config entries will be created for the projectors?

Reply:
it's creating new config_entry for most user on each restart right now, so I'd call it bugfix.